### PR TITLE
Polish code and comment of gpfdist (7X)

### PR DESCRIPTION
**Pull request by hyongtao-db**
There are 3 commits about polishing gpfdist in this pr.
Please review them one by one. Thanks.

**First commit afccfb48fa8ad1f47991fe335edfcd96a7ef04b5**
Apply standard C format code and comment. Such as:
- change comment from `//` to `/* */`
- change `if {` to 
```
if 
{
```
et. al

**Second commit 46bf222b12ed300960340b159ecf10d5c83132a6**
Reomve 2 unnecessary assignments, take one as an example
From
```
int res = recycle_thread(r);
if (res < 0)
```
To
```
if (recycle_thread(r) < 0)
```
 
**Third commit b3d84ecf702036c0f3fbca49b8826808e4871cc7**
- add `#define ERROR_CODE_SUCCESS 0` and `#define ERROR_CODE_GENERIC 1`
- rewrite `session_end(r->session, 0);` as `session_end(r->session, ERROR_CODE_SUCCESS);`
- rewrite `request_end(r, 0, 0);` as `request_end(r, ERROR_CODE_SUCCESS, 0);`

Signed-off-by: Yongtao Huang <yongtaoh@vmware.com>

### CODE REVIEW

**Comment by hyongtao-db** [comment]
All the test case passed. Please refer to https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_dev_hyt/jobs/icw_planner_rocky8/builds/1

**Comment by zhaorui20005** [comment]
It looks more comfortable. Thanks for the work.

File: src/bin/gpfdist/gpfdist.c [hunk 1]
```diff
@@ -70,6 +70,9 @@
 #define MAX_FRAME_SIZE 65536
 #define MAX_THREAD_NUM 256
 
+#define VALID_NUM 0
```
Comment (zhaorui20005) [inline src/bin/gpfdist/gpfdist.c:73]: Maybe change the VALID_NUM to another one. But I'm not good at naming things.

Comment (hyongtao-db) [reply src/bin/gpfdist/gpfdist.c:73]: > Maybe change the VALID_NUM to another one. But I'm not good at naming things.
Thanks. I am not adept at it too. HAHA. Please let me think it carefully.

Comment (hyongtao-db) [reply src/bin/gpfdist/gpfdist.c:73]: @zhaorui20005 @huiliang-liu So how about `OK_NUM`  or `SUCCESS_NUM` ？

Comment (huiliang-liu) [reply src/bin/gpfdist/gpfdist.c]: It is not good name either. what about changing them to `ERROR_CODE_SUCCESS` and `ERROR_CODE_GENERIC` ?

**Review by zhaorui20005 — APPROVED** [review]
LGTM

File: src/bin/gpfdist/gpfdist.c (hunk 1 above)
Comment (hyongtao-db) [reply src/bin/gpfdist/gpfdist.c]: Done.

**Review by huiliang-liu — APPROVED** [review]

**Pull request merged**